hack/build_proxy.sh

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+: "${ISTIO_GITHUB_REPO_URL:=https://github.com/maistra/istio.git}"
+: "${ISTIO_GITHUB_BRANCH:=maistra-3.0}"
+: "${PROXY_GITHUB_REPO_URL:=https://github.com/maistra/proxy.git}"
+: "${PROXY_GITHUB_BRANCH:=maistra-3.0-pre}"
+
+function download_envoy_binary() {
+  if [ $# -ne 1 ]; then
+    echo "Usage: download_envoy_binary workdir"
+    exit 1
+  fi
+
+  local workdir="${1}"
+
+  git clone --single-branch -b "${ISTIO_GITHUB_BRANCH}" "${ISTIO_GITHUB_REPO_URL}" "${workdir}/istio"
+  cd "${workdir}/istio"
+  make init
```

Comment:
I think by default this will run within the build container. We don't need that.

Either set `BUILD_WITH_CONTAINER=0` or don't run `make init`. If you look at bin/init.sh you will see it's quite trivial to get the proxy [sha](https://github.com/istio/istio/blob/master/bin/init.sh#L37) and download it from gcs directly. Up to you.